azurerm_mssql_server_vulnerability_assessment is failing without access key/sas signature

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

Hi, I have a storage account which I created specifically for this solution (Vulnerability assessment), which is behind both a virtual network and a firewall. However, in the terraform pipeline we're using the new AzureRM v4.31.0 provider version, which is failing with the following error:

**Error: 1 error occurred:
	* at least one of 'storage_account_access_key' or 'storage_container_sas_key' properties must be set, got nil, nil**

According to the official documentation, neither 'storage_account_access_key' or 'storage_container_sas_key' can be set when the storage account is behind such virtual network or firewall - which indicates that there is a bug in the AzureRM version.


<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.9.7

### AzureRM Provider Version

4.31.0 (latest)

### Affected Resource(s)/Data Source(s)

azurerm_mssql_server_vulnerability_assessment

### Terraform Configuration Files

```hcl
resource "azurerm_mssql_server_vulnerability_assessment" "example" {
  server_security_alert_policy_id = azurerm_mssql_server_security_alert_policy.example.id
  storage_container_path          = var.storage_container_path
  recurring_scans {
    enabled                   = true
    email_subscription_admins = true
    emails = [
      "email@example1.com",
      "email@example2.com"
    ]
  }
}
```

### Debug Output/Panic Output

```shell
resource "azurerm_mssql_server_vulnerability_assessment" "example" {

##[error]Terraform command 'plan' failed with exit code '1'.:  1 error occurred: 	* at least one of 'storage_account_access_key' or 'storage_container_sas_key' properties must be set, got nil, nil
##[error]
Error: 1 error occurred:
	* at least one of 'storage_account_access_key' or 'storage_container_sas_key' properties must be set, got nil, nil
```

### Expected Behaviour

should work without each  'storage_account_access_key' or 'storage_container_sas_key' argument :)

### Actual Behaviour

got panic nil errors

### Steps to Reproduce

terraform plan

### Important Factoids

_No response_

### References

1. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_server_vulnerability_assessment#Microsoft.Sql-1
2. Error is triggered due to this commit - https://github.com/hashicorp/terraform-provider-azurerm/pull/29373

--- a/internal/services/mssql/mssql_server_vulnerability_assessment_resource.go
+++ b/internal/services/mssql/mssql_server_vulnerability_assessment_resource.go
@@ -117,21 +117,21 @@ func resourceMsSqlServerVulnerabilityAssessmentCreate(d *pluginsdk.ResourceData,
 
 	alertResult, err := alertClient.Get(ctx, serverId)
 	if err != nil {
-		return fmt.Errorf("retrieving mssql server security alert policy: %+v", err)
+		return fmt.Errorf("retrieving security alert policy for SQL server %q: %w", serverId.ServerName, err)
 	}
 
 	model := alertResult.Model
 	if model == nil {
-		return fmt.Errorf("retrieving mssql server security alert policy %s: model was nil", serverId)
+		return fmt.Errorf("security alert policy for SQL server %q returned empty response", serverId.ServerName)
 	}
 
 	alertProps := model.Properties
 	if alertProps == nil {
-		return fmt.Errorf("retrieving mssql server security alert policy properties %s: properties was nil", serverId)
+		return fmt.Errorf("security alert policy for SQL server %q has no properties", serverId.ServerName)
 	}
 
 	if alertProps.State != serversecurityalertpolicies.SecurityAlertsPolicyStateEnabled {
-		return fmt.Errorf("mssql server security alert policy is not 'enabled'")
+		return fmt.Errorf("security alert policy for SQL server %q must be enabled before configuring vulnerability assessment", serverId.ServerName)
 	}
 
 	log.Printf("[INFO] preparing arguments for mssql server vulnerability assessment creation")
@@ -188,7 +188,7 @@ func resourceMsSqlServerVulnerabilityAssessmentCreate(d *pluginsdk.ResourceData,
 
 	result, err := client.CreateOrUpdate(ctx, serverId, payload)
 	if err != nil || result.Model == nil || result.Model.Id == nil {
-		return fmt.Errorf("creating mssql server vulnerability assessment %s : %v", id, err)
+		return fmt.Errorf("creating vulnerability assessment for SQL server %q: %w", serverId.ServerName, err)
 	}
 
 	d.SetId(id.ID())
@@ -214,27 +214,27 @@ func resourceMsSqlServerVulnerabilityAssessmentRead(d *pluginsdk.ResourceData, m
 	vulnerability, err := client.Get(ctx, serverId)
 	if err != nil {
 		if response.WasNotFound(vulnerability.HttpResponse) {
-			log.Printf("[WARN] mssql server vulnerability assessment %s: not found", id)
+			log.Printf("[WARN] vulnerability assessment for SQL server %q was not found", serverId.ServerName)
 			d.SetId("")
 			return nil
 		}
 
-		return fmt.Errorf("retrieving mssql server vulnerability assessment %s: %+v", *id, err)
+		return fmt.Errorf("retrieving vulnerability assessment for SQL server %q: %w", serverId.ServerName, err)
 	}
 
 	alert, err := alertClient.Get(ctx, serverId)
 	if err != nil {
-		return fmt.Errorf("retrieving mssql server security alert policy by ID: %+v", err)
+		return fmt.Errorf("retrieving security alert policy for SQL server %q: %w", serverId.ServerName, err)
 	}
 
 	model := alert.Model
 
 	if model == nil {
-		return fmt.Errorf("retrieving mssql server security alert policy %s: model was nil", serverId)
+		return fmt.Errorf("security alert policy for SQL server %q returned empty response", serverId.ServerName)
 	}
 
 	if model.Id == nil {
-		return fmt.Errorf("retrieving mssql server security alert policy %s: Id was nil", serverId)
+		return fmt.Errorf("security alert policy for SQL server %q has no resource ID", serverId.ServerName)
 	}
 
 	recurringScans := make([]interface{}, 0)
@@ -303,38 +303,37 @@ func resourceMsSqlServerVulnerabilityAssessmentUpdate(d *pluginsdk.ResourceData,
 
 	alert, err := alertClient.Get(ctx, serverId)
 	if err != nil {
-		return fmt.Errorf("retrieving mssql server security alert policy: %+v", err)
+		return fmt.Errorf("retrieving security alert policy for SQL server %q: %w", serverId.ServerName, err)
 	}
 
 	if alert.Model == nil {
-		return fmt.Errorf("retrieving mssql server security alert policy %s: model was nil", serverId)
+		return fmt.Errorf("security alert policy for SQL server %q returned empty response", serverId.ServerName)
 	}
 
 	if alert.Model.Properties == nil {
-		return fmt.Errorf("retrieving mssql server security alert policy properties %s: properties was nil", serverId)
+		return fmt.Errorf("security alert policy for SQL server %q has no properties", serverId.ServerName)
 	}
 
 	if alert.Model.Properties.State != serversecurityalertpolicies.SecurityAlertsPolicyStateEnabled {
-		return fmt.Errorf("mssql server security alert policy is not 'enabled'")
+		return fmt.Errorf("security alert policy for SQL server %q must be enabled before updating vulnerability assessment", serverId.ServerName)
 	}
 
 	log.Printf("[INFO] preparing arguments for mssql server vulnerability assessment update")
 
 	payload := servervulnerabilityassessments.ServerVulnerabilityAssessment{}
-
 	result, err := client.Get(ctx, serverId)
 	if err != nil {
-		return fmt.Errorf("retrieving mssql server vulnerability assessment policy: %+v", err)
+		return fmt.Errorf("retrieving existing vulnerability assessment for SQL server %q: %w", serverId.ServerName, err)
 	}
 
 	if result.Model == nil {
-		return fmt.Errorf("retrieving mssql server vulnerability assessment policy %s: model was nil", serverId)
+		return fmt.Errorf("vulnerability assessment for SQL server %q returned empty response", serverId.ServerName)
 	}
 
 	props := result.Model.Properties
 
 	if props == nil {
-		return fmt.Errorf("retrieving mssql server vulnerability assessment policy %s: properties was nil", serverId)
+		return fmt.Errorf("vulnerability assessment for SQL server %q has no properties", serverId.ServerName)
 	}
 
 	if d.HasChange("storage_container_path") {
@@ -394,7 +393,7 @@ func resourceMsSqlServerVulnerabilityAssessmentUpdate(d *pluginsdk.ResourceData,
 
 	update, err := client.CreateOrUpdate(ctx, serverId, payload)
 	if err != nil || update.Model == nil || update.Model.Id == nil {
-		return fmt.Errorf("updating mssql server vulnerability assessment %s : %v", id, err)
+		return fmt.Errorf("updating vulnerability assessment for SQL server %q: %w", serverId.ServerName, err)
 	}
 
 	return resourceMsSqlServerVulnerabilityAssessmentRead(d, meta)
@@ -415,7 +414,7 @@ func resourceMsSqlServerVulnerabilityAssessmentDelete(d *pluginsdk.ResourceData,
 	serverId := commonids.NewSqlServerID(id.SubscriptionId, id.ResourceGroup, id.ServerName)
 
 	if _, err = client.Delete(ctx, serverId); err != nil {
-		return fmt.Errorf("deleting mssql server vulnerability assessment %s: %v", *id, err)
+		return fmt.Errorf("deleting vulnerability assessment for SQL server %q: %w", serverId.ServerName, err)
 	}
 
 	return nil

